src/emulator/functionsRuntimeWorker.ts

```diff
@@ -110,8 +127,10 @@ export class RuntimeWorker {
     });
   }
 
-  waitForSystemLog(filter: (el: EmulatorLog) => boolean): Promise<EmulatorLog> {
-    return EmulatorLog.waitForLog(this.runtime.events, "SYSTEM", "runtime-status", filter);
+  waitForSocketReady(): Promise<any> {
```

Comment:
If the calling site does not immediately call this method right after constructor, there will still be a race-condition. I'd suggest inlining this into the constructor (like `this.socketReady = ...`).

Reply:
Ah good call!  I will do it on the transition to BUSY